credentials/credentials_ext_test.go

```diff
@@ -0,0 +1,338 @@
+	if err != nil {
+		log.Fatalf("Failed to create credentials %v", err)
+	}
+}
+
+func authorityChecker(ctx context.Context, expectedAuthority string) (*testpb.Empty, error) {
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		return nil, status.Error(codes.InvalidArgument, "failed to parse metadata")
+	}
+	auths, ok := md[":authority"]
+	if !ok {
+		return nil, status.Error(codes.InvalidArgument, "no authority header")
+	}
+	if len(auths) != 1 {
+		return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("no authority header, auths = %v", auths))
+	}
+	if auths[0] != expectedAuthority {
+		return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("invalid authority header %v, expected %v", auths[0], expectedAuthority))
+	}
+	return &testpb.Empty{}, nil
+}
+
+func checkUnavailableRPCError(t *testing.T, err error) {
+	t.Helper()
+	if err == nil {
+		t.Fatalf("EmptyCall() should fail")
+	}
+	s, ok := status.FromError(err)
+	if !ok {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	if s.Code() != codes.Unavailable {
+		t.Fatalf("EmptyCall() = _, %v, want _, error code: %v", s.Code(), codes.Unavailable)
+	}
+}
+
+// Tests the grpc.CallAuthority option with TLS credentials. This test verifies
+// that the provided authority is correctly propagated to the server when using TLS.
+// It covers both positive and negative cases:  correct authority and incorrect
+// authority, expecting the RPC to fail with `UNAVAILABLE` status code error in
+// the latter case.
+func (s) TestAuthorityCallOptionsWithTLSCreds(t *testing.T) {
+	tests := []struct {
+		name         string
+		expectedAuth string
+		wantRPCError bool
+	}{
+		{
+			name:         "CorrectAuthorityWithTLS",
+			expectedAuth: "auth.test.example.com",
+			wantRPCError: false,
+		},
+		{
+			name:         "IncorrectAuthorityWithTLS",
+			expectedAuth: "auth.example.com",
+			wantRPCError: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ss := &stubserver.StubServer{
+				EmptyCallF: func(ctx context.Context, _ *testpb.Empty) (*testpb.Empty, error) {
+					return authorityChecker(ctx, tt.expectedAuth)
+				},
+			}
+			if err := ss.StartServer(grpc.Creds(credentials.NewServerTLSFromCert(&cert))); err != nil {
+				t.Fatalf("Error starting endpoint server: %v", err)
+			}
+			defer ss.Stop()
+
+			cc, err := grpc.NewClient(ss.Address, grpc.WithTransportCredentials(creds))
+			if err != nil {
+				t.Fatalf("grpc.NewClient(%q) = %v", ss.Address, err)
+			}
+			defer cc.Close()
+
+			ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+			defer cancel()
+
+			_, err = testgrpc.NewTestServiceClient(cc).EmptyCall(ctx, &testpb.Empty{}, grpc.CallAuthority(tt.expectedAuth))
+			if tt.wantRPCError {
+				checkUnavailableRPCError(t, err)
+			} else if err != nil {
+				t.Fatalf("EmptyCall() rpc failed: %v", err)
+			}
+		})
+	}
+}
+
+func (s) TestTLSCredsWithNoAuthorityOverride(t *testing.T) {
```

Comment:
Please add a test comment for this.

What scenario is this testing? If this test fails, what does it indicate about the authority override feature?